conformance/conformance_test.go

```diff
@@ -45,15 +46,37 @@ func TestConformance(t *testing.T) {
 
 	t.Logf("Running conformance tests with %s GatewayClass", *flags.GatewayClassName)
 
+	supportedFeatures := parseSupportedFeatures(*flags.SupportedFeatures)
+	exemptFeatures := parseExemptFeatures(*flags.ExemptFeatures)
+
 	cSuite := suite.New(suite.Options{
 		Client:               client,
 		GatewayClassName:     *flags.GatewayClassName,
 		Debug:                *flags.ShowDebug,
 		CleanupBaseResources: *flags.CleanupBaseResources,
-		SupportedFeatures: []suite.SupportedFeature{
-			suite.SupportReferenceGrant,
```

Comment:
I believe the original intent of having this specific feature hardcoded was to include ReferenceGrant in the default set of conformance tests - I'm not sure if we should try to keep a "default supported features" list somehow (feels awkward), or instead just change conformance tests currently enabled with the flag to instead switch to an ExemptFeature flag to opt-out.

Reply:
Maybe we can make a "default supported features" which runs all the conformance tests, and we can change enabled tests with `SupportedFeatures` flag or just opt-out tests with `ExemptFeatures` flag. What do you think?

Reply:
I think we could just remove the [`suite.SupportReferenceGrant`](https://github.com/kubernetes-sigs/gateway-api/search?q=SupportReferenceGrant) constant, and remove it from the `Features` list on all conformance tests that currently expect it to be set.

This keeps those tests included by default, still allows using the `ExemptReferenceGrant` flag to opt-out, avoids any complexity of an extra "default" list, and feels okay given that we're intending to promote ReferenceGrant to v1beta1 in the upcoming v0.6.0 Gateway API release.

Reply:
@mikemorris Thanks for your reply, I can understand it basically. What to do are:
1. Remove [suite.SupportReferenceGrant](https://github.com/kubernetes-sigs/gateway-api/search?q=SupportReferenceGrant) as the default value of `SupportedFeatures`
2. Remove [suite.SupportReferenceGrant](https://github.com/kubernetes-sigs/gateway-api/search?q=SupportReferenceGrant) from `Features` list on all tests, such as https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/tests/gateway-secret-invalid-reference-grant.go#L37. Then replace it to ```Exemptions: []suite.ExemptFeature{
		suite.ExemptReferenceGrant,
	},```, so that `--exempt-features ExemptReferenceGrant` flag can be set to opt-out.
	
Can I confirm it is correct?

Reply:
Yes, that should be correct @gyohuangxin!

I think we can then do
3. Remove the constant at https://github.com/kubernetes-sigs/gateway-api/blob/583e4e2541b757d5e72d23c18f2d7201b49bccf2/conformance/utils/suite/suite.go#L47-L48. In upcoming Gateway API v0.6.0, all implementations should assume ReferenceGrant is supported unless explicitly setting `ExemptReferenceGrant`.

Reply:
@mikemorris Thanks! I've updated the code and the "user-facing change" section per your comments, please review again.